ikarus/io/resultfunction.hh

```diff
@@ -109,34 +109,40 @@ public:
    * \brief Get the precision used for this result
    * \details
    * This function is part of the Dune::VTKFunction interface.
-   * This has no affect when the ResultFunction is used as part of a the dune-vtk module
+   * This has no affect when the ResultFunction is used with the dune-vtk module
    *
    * \return Precision (i.e. float64 or float32)
    */
   Dune::VTK::Precision precision() const override { return prec_; }
 
   /**
-   * \brief Constructor for ResultFunction.
-   * \details
-   * Constructs a ResultFunction object with given finite elements, ferequirements
+   * \brief Construct a new Result Function object
    *
+   * \tparam Args Type of additional arguments passed to the user function
    * \param assembler shared pointer to the underlying assembler (provides the finite elements and the requested
    * results)
    * \param prec (optional) specify the used precision (only has an effect when using resultfunciton with
    * Dune::VTK::VTKWriter)
+   * \param args additional arguments passed to the constructor of the user function
    */
-  ResultFunction(std::shared_ptr<Assembler> assembler, Dune::VTK::Precision prec = Dune::VTK::Precision::float64)
+  template <typename... Args>
+  ResultFunction(std::shared_ptr<Assembler> assembler, Dune::VTK::Precision prec = Dune::VTK::Precision::float64,
```

Comment:
See below the suggestion of templating this with the user function instead of the args